test/auth_cluster/test_auth_v2_migration.py

```diff
@@ -163,3 +163,65 @@ async def test_auth_v2_migration(request, manager: ManagerClient):
 
     logging.info("Checking auth statements after migration")
     await check_auth_v2_works(manager, hosts)
+
+
+@pytest.mark.asyncio
+async def test_auth_v2_during_recovery(manager: ManagerClient):
```

Comment:
The test should also check that the old auth data are not migrated when we re-run upgrade on recovery. For example, you could:

1. Start the cluster in legacy mode (via error injection)
2. Create some roles
3. Upgrade the cluster to raft topology
4. Drop a role in the auth-v2 mode
5. Perform recovery
6. Observe that the role that was deleted in point 4 wasn't resurrected

Reply:
I'd prefer to not change this test to start with v1 because then it gets very similar to the test above it and also more complicated. I could create v1 role by insert to old table, trying the "poison" approach